WcaOnRails/app/controllers/application_controller.rb

```diff
@@ -37,6 +55,15 @@ def doorkeeper_unauthorized_render_options(error: nil)
     devise_parameter_sanitizer.for(:account_update) << :name << :email
   end
 
+  # This method is called by devise after a successful login to know the redirect path
+  # We override it to do some action after signing in, but we want to use the original path
+  protected def after_sign_in_path_for(resource)
+    # When the user signs in, 'session[:locale]' is not cleared, so we need to update
+    # it with the user's preferred_locale
+    session[:locale] = current_user.preferred_locale || I18n.default_locale
```

Comment:
Hmm.. what if the current user does not have a `preferred_locale` set in the database, but their `Accept-Language` http header is set to something other than english? It seems to me that we should be obeying their `Accept-Language` header in this case. I feel like this is a duplicate of the already well written logic in the `set_locale` method, could we just call `set_locale` here instead?